src/goto-programs/json_goto_trace.cpp

```diff
@@ -94,6 +96,39 @@ void convert(
           step.full_lhs.is_not_nil(),
           "full_lhs in assignment must not be nil");
         exprt simplified=simplify_expr(step.full_lhs, ns);
+
+        if(trace_options.json_full_lhs)
+        {
+          class comment_base_name_visitort: public expr_visitort
+          {
+          private:
+            const namespacet &ns;
+
+          public:
+            explicit comment_base_name_visitort(const namespacet &ns) : ns(ns)
+            { }
+
+            void operator()(exprt &expr) override
+            {
+              if(expr.id() == ID_symbol)
+              {
+                const symbolt &symbol = ns.lookup(expr.get(ID_identifier));
+                // Don't break sharing unless need to write to it
+                const irept::named_subt &comments = expr.get_comments();
```

Comment:
Sharing will be broken in multiple ways: `expr` is non-const, thus a non-const member will be called; and sharing has been broken at the prior level already by invoking `visit()`.